### PR TITLE
Implemented `XLMRoberta` sub-tasks

### PR DESCRIPTION
**Pull request by Seun-Ajayi**
## Proposed changes

Implemented `XLMRobertaForSequenceClassification`, `XLMRobertaForTokenClassification` and XLMRobertaForQuestionAnswering and provided their corresponding tests. All tests passed. 

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] New Model Architecture
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

**Comment by ToluClassics** [comment]
Sweet!

**Comment by ToluClassics** [comment]
Can you run install and run pre-commit on your files

File: src/mlx_transformers/models/xlm_roberta.py [hunk 1]
```diff
@@ -451,6 +451,235 @@ def __call__(
...
+                start_positions = start_positions.squeeze(-1)
+            if len(end_positions.size()) > 1:
+                end_positions = end_positions.squeeze(-1)
+            # sometimes the start/end positions are outside our model inputs,
+            # we ignore these terms
+            ignored_index = start_logits.size(1)
+            start_positions = start_positions.clamp(0, ignored_index)
+            end_positions = end_positions.clamp(0, ignored_index)
+
+            loss_fct = nn.losses.cross_entropy()
```
Comment (ToluClassics) [inline src/mlx_transformers/models/xlm_roberta.py]: `nn.losses.cross_entropy` is a function so it should be ` loss_fct = nn.losses.cross_entropy`

File: src/mlx_transformers/models/xlm_roberta.py [hunk 2]
```diff
@@ -451,6 +451,235 @@ def __call__(
...
+        sequence_output = outputs.last_hidden_state
+
+        sequence_output = self.dropout(sequence_output)
+        logits = self.classifier(sequence_output)
+
+        loss = None
+        if labels is not None:
+            # move labels to correct device to enable model parallelism
+            labels = labels.to(logits.device)
+            loss_fct = nn.losses.cross_entropy()
```
Comment (ToluClassics) [inline src/mlx_transformers/models/xlm_roberta.py]: `nn.losses.cross_entropy` is a function so it should be ` loss_fct = nn.losses.cross_entropy`

File: src/mlx_transformers/models/xlm_roberta.py [hunk 3]
```diff
@@ -451,6 +451,235 @@ def __call__(
...
+                if self.num_labels == 1:
+                    loss = loss_fct(logits.squeeze(), labels.squeeze())
+                else:
+                    loss = loss_fct(logits, labels)
+            elif self.config.problem_type == "single_label_classification":
+                loss_fct = nn.losses.cross_entropy()
+                loss = loss_fct(
+                    logits.view(-1, self.num_labels), labels.view(-1))
+            elif self.config.problem_type == "multi_label_classification":
+                loss_fct = nn.losses.binary_cross_entropy()
```
Comment (ToluClassics) [inline src/mlx_transformers/models/xlm_roberta.py]: same

File: src/mlx_transformers/models/xlm_roberta.py [hunk 4]
```diff
@@ -451,6 +451,235 @@ def __call__(
...
+                    self.config.problem_type = "multi_label_classification"
+
+            if self.config.problem_type == "regression":
+                loss_fct = nn.losses.mse_loss()
+                if self.num_labels == 1:
+                    loss = loss_fct(logits.squeeze(), labels.squeeze())
+                else:
+                    loss = loss_fct(logits, labels)
+            elif self.config.problem_type == "single_label_classification":
+                loss_fct = nn.losses.cross_entropy()
```
Comment (ToluClassics) [inline src/mlx_transformers/models/xlm_roberta.py]: same

File: src/mlx_transformers/models/xlm_roberta.py [hunk 5]
```diff
@@ -451,6 +451,235 @@ def __call__(
...
+                    self.config.problem_type = "regression"
+                elif self.num_labels > 1 and (
+                    labels.dtype == mx.long or labels.dtype == mx.int
+                    ):
+                    self.config.problem_type = "single_label_classification"
+                else:
+                    self.config.problem_type = "multi_label_classification"
+
+            if self.config.problem_type == "regression":
+                loss_fct = nn.losses.mse_loss()
```
Comment (ToluClassics) [inline src/mlx_transformers/models/xlm_roberta.py]: same

File: src/mlx_transformers/models/xlm_roberta.py [hunk 6]
```diff
@@ -451,6 +451,235 @@ def __call__(
         )
 
 
+class XLMRobertaClassificationHead(nn.Module):
```
Comment (ToluClassics) [inline src/mlx_transformers/models/xlm_roberta.py:455]: I created a Mixin to allow us load directly from Huggingface without having to convert first, so we can change add it here by importing the mixin

`from .base import MlxPretrainedMixin` and then `class XLMRobertaClassificationHead(nn.Module, MlxPretrainedMixin):`

Comment (ToluClassics) [reply src/mlx_transformers/models/xlm_roberta.py:455]: With this we can easily do 

```
model = XLMRobertaClassificationHead(config)
model.from_pretrained(model_name)
```

Comment (Seun-Ajayi) [reply src/mlx_transformers/models/xlm_roberta.py:455]: perfect, this affected the NSP I was building for Bert. So I paused it to fix the Pretrained class first

File: src/mlx_transformers/models/xlm_roberta.py (hunk 5 above)
Comment (Seun-Ajayi) [reply src/mlx_transformers/models/xlm_roberta.py]: This would apply to roberta and bert too, right?

Comment (ToluClassics) [reply src/mlx_transformers/models/xlm_roberta.py]: Yes

Comment (Seun-Ajayi) [reply src/mlx_transformers/models/xlm_roberta.py]: My bad, the naming convention should have given it away

**Comment by Seun-Ajayi** [comment]
> Can you run install and run pre-commit on your files

Used autopep8 before now, would use pre-commit now

**Comment by Seun-Ajayi** [comment]
Ready to be Merged

**Review by ToluClassics — APPROVED** [review]

**Pull request merged**